spec/index.bs

```diff
@@ -755,11 +758,14 @@ This specification defines the following [=usage scopes=]:
     - Resources used in bind group entries with visibility 0, or visible only to the the compute stage but used in a render pass or vice versa.
 </div>
 
-The [=usage scopes=] are validated at {{GPUCommandEncoder/finish()|GPUCommandEncoder.finish}} time.
-The implementation performs the <dfn dfn>usage scope validation</dfn> by composing
-the list of all [=internal usage=] flags of each [=subresource=] used in the [=usage scope=].
-A {{GPUValidationError}} is generated in the current scope with an appropriate error message
-if that list is not a [=compatible usage list=].
+During command encoding, every usage of a subresource is recorded in one of the
```

Comment:
I suppose for `dispatch()` a usage becomes recorded in multiple usage scopes, so this needs to be adjusted here

Reply:
Hmm, do you mean that usages in bindings are inherited across dispatch() if they are not replaced by another setBindGroup? @kvark

Reply:
yes, a single subresource usage, if coming from `setBindGroup`, may affect multiple dispatches, which have their own usage scopes.

Reply:
Agreed. Alternatively, we can say that (for example, via a `Note') bindings will be inherited across dispatch calls until they are replaced by another setBindGroup call, but they can't be inherited across compute pass. And we can also say that replaced bindings should not be validated in compute pass.

Reply:
We were just discussing this on the editors meeting yesterday, and the conclusion was to say that each `dispatch` captures a bunch of context into its usage scope. I believe @kainino0x is working on specifying this

Reply:
My original concern still stands. This shouldn't say "in one of the" but instead imply one more more scopes.

Reply:
It doesn't make sense to me to specify it that way, because when you say "one or more scopes" you aren't saying which scopes. Instead we say the dispatch looks at the current table of bind groups and adds all of the bind groups' subresources to the usage scope.

Reply:
I'm not suggesting to say "one or more scopes" (of course that would be unclear), I suggested to write *something* that *implies* that.
> Instead we say the dispatch looks at the current table of bind groups and adds all of the bind groups' subresources to the usage scope.

Sure. The original sentence that spawned this thread still needs fixing.

Reply:
looks like it still says "one of the"?

Reply:
"Every usage of a subresource" (in compute, this is exactly dispatches and nothing else, doesn't include setBindGroup which doesn't constitute a usage) "is recorded in one of the usage scopes in the command buffer."

Reply:
I guess it's just not obvious that `setBindGroup` is not "using" resources, but for the sake of the spec it's probably fine.

Reply:
It's said explicitly, though:

> In a compute pass, ...
> State-setting compute pass commands, like
> {{GPUProgrammablePassEncoder/setBindGroup(index, bindGroup, dynamicOffsets)}},
> do not contribute directly to a usage scope; they instead change the
> state that is checked in dispatch commands.